src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -35,6 +37,13 @@ public override ExportResult Export(in Batch<T> batch)
                 return ExportResult.Failure;
             }
 
+            // Note: this mitigates the growing ExportedItems issue.
+            // Need to discuss if this is an acceptible change.
+            if (this.isMetric)
+            {
+                this.exportedItems.Clear();
```

Comment:
Let's first decide on this. Do we want the collection of exported items to hold the `MetricPoint` state corresponding to every export made so far?

Reply:
Personally, I feel like... no, it shouldn't auto-clear anything. Whoever constructed the `InMemoryExporter` owns the collection that was passed in. So they can clear it if needed based on whatever test is being modeled? But yes the exporter should move copies into that collection so the data doesn't continue to change.

Reply:
Hi @CodeBlanch, 
If the InMemoryExporter could deepcopy the `Metric` into the export collection, I believe that would fully address this issue.
However, I don't see an obvious way to accomplish this.
I listed [here](https://github.com/open-telemetry/opentelemetry-dotnet/issues/2361#issuecomment-1042421246) several methods to perform a deepcopy and the blocking issues with each.
If you have any other ideas, I'm eager to get your input!